packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -39,6 +62,86 @@ class OverScrollHeaderStretchConfiguration {
   final AsyncCallback onStretchTrigger;
 }
 
+/// {@template flutter.rendering.persistentHeader.showOnScreenConfiguration}
+/// Specifies how a pinned header or a floating header should react to
+/// [RenderObject.showOnScreen] calls.
+/// {@endtemplate}
+@immutable
+class PersistentHeaderShowOnScreenConfiguration {
+  /// Creates an object that specifies how a pinned or floating persistent header
+  /// should behave in response to [RenderObject.showOnScreen] calls.
+  const PersistentHeaderShowOnScreenConfiguration({
+    this.ignoreLeading = true,
+    this.minShowOnScreenExtent,
+    this.maxShowOnScreenExtent,
+  }) : assert(ignoreLeading != null),
+       assert(minShowOnScreenExtent == null || maxShowOnScreenExtent == null || minShowOnScreenExtent <= maxShowOnScreenExtent);
+
+  /// {@template flutter.rendering.persistentHeader.ignoreLeading}
+  /// Whether to ignore the part of the `rect` specified in [RenderObject.showOnScreen]
+  /// that exceeds the leading edge of the sliver's child [RenderBox], to prevent
+  /// `showOnScreen` from significantly reduce the viewport's scroll offset in
+  /// an attempt to reveal to original `rect`.
+  ///
+  /// When this parameter is set to false, the framework will try to reveal the
+  /// entire `rect` specified in [RenderObject.showOnScreen], even if it exceeds
+  /// the leading edge of the persistent header. As a result, the viewport will
+  /// attempt to reveal the part of the preceeding sliver(s) that intersects with
+  /// `rect`. This could potentially lead to unintended overscrolling if there's
+  /// no preceeding slivers. Additionally, if the sliver is initially pinned
+  /// (or floating) at the leading edge, this could also reduce the scroll
+  /// offset of the viewport significantly, revealing the slivers that are
+  /// immediately after the pinned header.
+  ///
+  /// Setting the parameter to true will cause the framework to trim the incoming
+  /// `rect` in `showOnScreen` if it exceeds the leading edge of the `child`
+  /// [RenderBox]. The new `rect` will have the same leading edge as the `child`
+  /// [RenderBox], or that of the persistent header when `child` is null.
+  /// {@endtemplate}
+  ///
+  /// Defaults to true and must not be null.
+  final bool ignoreLeading;
+
+  /// {@template flutter.rendering.persistentHeader.minShowOnScreenExtent}
+  /// The smallest the floating header can expand to in the main axis direction,
+  /// when `showOnScreen` is called, in addition to the persistent header's
+  /// `minExtent`.
```

Comment:
What `minExtent are you refering to here? Is this https://master-api.flutter.dev/flutter/widgets/SliverPersistentHeaderDelegate/minExtent.html? Maybe make the doc link to it properly. Same for maxExtent